openpgp/s2k/s2k.go

```diff
@@ -168,11 +175,26 @@ func parseGNUExtensions(r io.Reader) (f func(out, in []byte), err error) {
 		return
 	}
 	gnuExtType := int(buf[0])
-	if gnuExtType != 1 {
-		return nil, errors.UnsupportedError("unknown S2K GNU protection mode: " + strconv.Itoa(int(gnuExtType)))
-	}
+	if gnuExtType == 1 {
+		return nil, nil
+	} else if gnuExtType == 2 {
+		var lenBuf [1]byte
+		_, err = io.ReadFull(r, buf[:1])
+		if err != nil {
+			return
+		}
+
+		var ivBuf [16]byte
+		ivLen := minInt(lenBuf[0], 16)
```

Comment:
Ah, you're right. It's a `min`.  I'll merge and then tweak the style a bit, thanks!